akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -631,7 +636,6 @@ abstract class ShardCoordinator(
   var state = State.empty.withRememberEntities(settings.rememberEntities)
   // rebalanceInProgress for the ShardId keys, pending GetShardHome requests by the ActorRef values
   var rebalanceInProgress = Map.empty[ShardId, Set[ActorRef]]
-  var rebalanceWorkers: Set[ActorRef] = Set.empty
```

Comment:
It seems that the only reason why we keep this Set is to send a `ShardRegionTerminated` to the worked in case the region terminates. 

So, I'm wondering if it doesn't make the code simpler if we let the worker watch the region instead. As such we have less logic on the `ShardCoordinator` side and we don't have to keep that around. 

Is there any drawback in doing so? It increases the number of watchers and I don't know if that's something we can neglect or not.

Reply:
Sounds like it makes things cleaner and I don't see any obvious problem with it.

Reply:
The reason why the `HandOffWorker` isn't watching itself is for performance. The ShardCoordinator is anyway watching all regions, so it's better to have that single watch than each HandOffWorker establishes new watch with all regions. Watching includes message exchange, and it will also unwatch with new messages when it's stopped.